### ISSUE TITLE
Introduce shader examples that don't use high level rendering apis

### ISSUE THREAD

**Issue by IceSentry** [bevyengine/bevy#5843]
# Objective

Most of the shader examples use the `Material` trait and the few remaining ones that don't use it might be updated in the future to either use that or some other high level rendering api. 

We need an example that showcases a manual implementation for drawing a basic mesh without any high level apis.

There should also be an example showcasing a manual implementation of `AsBindGroup`.

These examples should be clearly used to showcase those lower level features and not be updated to higher level apis in the future.

# Notes

Discord dicussion: https://discord.com/channels/691052431525675048/743663924229963868/1014539753854615662

**Comment by IceSentry** [comment]
I think the example itself should be really simple. Nothing more complicated than the current shader_material example. The important part is that it showcases the mid-level rendering apis explicitly. Personally, I was planning on pretty much just writing the shader_material example but without using a Material. I just didn't have time yet.

**Comment by Vrixyz** [comment]
Basically re-establish a previous iteration of that example: https://github.com/bevyengine/bevy/blob/adb3ad399cf092110b2d69f7dc189abc0ebc9f80/examples/shader/shader_material.rs should might give a good base

**Comment by IceSentry** [comment]
Yeah, something like that would be a decent starting point. I think the important part would be having a lot of comment explaining what is going on.
